Woche4/filter-more/helpers.c

```diff
@@ -0,0 +1,183 @@
+
+            RGBTRIPLE imageTriple = image[k][i];
+
+            if (color_position == 0) {
+                sum += imageTriple.rgbtRed;
+            } else if (color_position == 1) {
+                sum += imageTriple.rgbtGreen;
+            } else {
+                sum += imageTriple.rgbtBlue;
+            }
+            counter++;
+
+        }
+    }
+
+    return round(sum /counter);
+}
+
+void blur(int height, int width, RGBTRIPLE image[height][width]) {
+    RGBTRIPLE copy[height][width];
+
+    for (int i = 0; i < height; i++) {
+        for (int j = 0; j < width; j++) {
+            copy[i][j] = image[i][j];
+        }
+    }
+
+    for (int i = 0; i < height; i++) {
+        for (int j = 0; j < width; j++) {
+            image[i][j].rgbtRed = getBlur(i, j, height, width, copy, 0);
+            image[i][j].rgbtGreen = getBlur(i, j, height, width, copy, 1);
+            image[i][j].rgbtBlue = getBlur(i, j, height, width, copy, 2);
+        }
+    }
+}
+
+void edges(int height, int width, RGBTRIPLE image[height][width]) {
+    RGBTRIPLE temp[height][width];
+         
+    int GxR;
+    int GyR;
+    int GxG;
+    int GyG;
+    int GxB;
+    int GyB;
+        
+    int Gx[3][3] = {{-1, 0, 1}, {-2, 0, 2}, {-1, 0, 1}};
+    int Gy[3][3] = {{-1, -2, -1}, {0, 0, 0}, {1, 2, 1}};
+        
+    for (int i = 0; i < height; i++) {
+        for (int j = 0; j < width; j++) {
+            GxR = 0;
+            GyR = 0;
+            GxG = 0;
+            GyG = 0;
+            GxB = 0;
+            GyB = 0;
+    
+            for (int h = -1; h < 2; h++) {
+                for (int w = -1; w < 2; w++) {
+                    if (i + h < 0 || i + h > height - 1) continue;
+                        
+                    if (j + w < 0 || j + w > width - 1) continue;
+
+                    RGBTRIPLE imageTriple = image[i + h][j + w];
+
+                    int GxValue = Gx[h + 1][w + 1];
+                    int GyValue = Gy[h + 1][w + 1];
+
+                    GxR += imageTriple.rgbtRed * GxValue;
+                    GxG += imageTriple.rgbtGreen * GxValue;
+                    GxB += imageTriple.rgbtBlue * GxValue;
+
+                    GyR += imageTriple.rgbtRed * GyValue;
+                    GyG += imageTriple.rgbtGreen * GyValue;
+                    GyB += imageTriple.rgbtBlue * GyValue;
+                }
+            }
+            
+            temp[i][j].rgbtRed = round(sqrt((GxR * GxR  + GyR * GyR)));
+            temp[i][j].rgbtGreen = round(sqrt((GxG * GxG + GyG * GyG)));
+            temp[i][j].rgbtBlue = round(sqrt((GxB * GxB + GyB * GyB)));
+                
+            if (temp[i][j].rgbtRed > 255) {
+                temp[i][j].rgbtRed = 255;
+            }
+                
+            if (temp[i][j].rgbtGreen > 255) {
+                temp[i][j].rgbtGreen = 255;
+            }
+                
+            if (temp[i][j].rgbtBlue > 255) {
+                temp[i][j].rgbtBlue = 255;
+            }
```

Comment:
This is similar to what you have done in your sepia-filter in line 36 to 38. Therefore you should make use of your limit function in this case, too, instead of re-implementing the same logic.
```suggestion
            temp[i][j].rgbtRed = filter(round(sqrt((GxR * GxR  + GyR * GyR))));
            temp[i][j].rgbtGreen = filter(round(sqrt((GxG * GxG + GyG * GyG))));
            temp[i][j].rgbtBlue = filter(round(sqrt((GxB * GxB + GyB * GyB))));
```